src/mgr/Mgr.cc

```diff
@@ -119,7 +121,10 @@ class MetadataUpdate : public Context
         if (daemon_state.exists(key)) {
           state = daemon_state.get(key);
 	  Mutex::Locker l(state->lock);
+          if (key.first == "mds")
           daemon_meta.erase("name");
```

Comment:
Need to indent this line now that it's inside an `if` (adding some `{}` would also be nice)